src/System/FSNotify/Types.hs

```diff
@@ -69,6 +69,9 @@ data WatchConfig = WatchConfig
   , confUsePolling :: Bool
     -- ^ Force use of polling, even if a more effective method may be
     -- available. This is mostly for testing purposes.
+  , confThreadPerEvent :: Bool
+    -- ^ If separate thread should be created for each event
```

Comment:
Nit: reword a little. How about "Fork a separate thread for each event when executing the user supplied event handler"

Reply:
Yes, that sounds better